src/renderers/PartialViewRenderers/region/facilitiesInfo.js

```diff
@@ -1,28 +1,42 @@
+const FUNCTIONAL = 'functional';
+const NON_FUNCTIONAL = 'non-functional';
+
 $(function () {
-    //get all facility related info
+
     getFacilitiesInfo();
-    //handler for showing or hiding all facilities
-    openCloseAllFacilities_handler();
-    //handler for facility  add
-    addFacility_handler();
-    //handle delete facility
-    deleteFacility_handler();
+
+    openCloseAllFacilities();
+
+    addFacility();
+
+    deleteFacility();
+
+    addFilterListener()
 });
 
 function getFacilitiesInfo() {
+
     let facilityIds = [];
     ipcRenderer.send('Facility:getFacilitiesByRegion', parseInt(window.process.argv.slice(-1)));
     ipcRenderer.once('Facility:getFacilitiesByRegionOK', (e, res) => {
         dataAvailable = true;
         if (res != null) {
+            $("#facilityList").empty();
+            let functionalFacilities = $('input[name=functionalFacilitiesDisplay]:checked').val();
+
+            if (functionalFacilities === FUNCTIONAL) {
+                res = res.filter(facility => facility.isFunctional)
+            } else if (functionalFacilities === NON_FUNCTIONAL) {
```

Comment:
How can `functionalFacilities` be `NON_FUNCTIONAL`? What does `$('input[name=functionalFacilitiesDisplay]:checked').val();` actually do?

Reply:
It basically checks the value of a group of radio buttons for what value it is. I admit the functionalFacilities is also followign the ones in components where it is usedComponents. I guess i should change he name to something clearer, like filterInput

Reply:
Ah, I think `facilityFilter is good, think `if (facilityFilter === FUNCTIONAL)`

Reply:
Alright Fixed.